### ISSUE TITLE
Enhancement: [rule-name] <a short description of my proposal>

### ISSUE THREAD

**Issue by glentakahashi** [typescript-eslint/typescript-eslint#7950]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/no-misused-promises/

### Description

no-misused-promises should also check if you try and do a nullish comparison on a Promise that is always defined, but where its value may be null

### Fail

```typescript
const test: Promise<string | null> = Promise.resolve(null);
if(test == null) {
  console.log('our promise value is not null'); // this is wrong, test value is null
}
```

### Pass

```typescript
const test: Promise<string | null> = Promise.resolve(null);
if(await test == null) {
  console.log('our promise value is not null');
}
```

**Comment by Josh-Cena** [comment]
This is already handled by `no-unnecessary-condition`, FYI. TBH, the scope of this rule is so broad that I'm not sure if this check belongs here or not, but it certainly feels weird to have overlapping reports from two rules.

**Comment by bradzacher** [comment]
I'm of two minds here yeah because the rule already has [significant duplication with `no-unnecessary-condition`](https://typescript-eslint.io/play#ts=5.3.0-beta&fileType=.ts&code=MYewdgzgLgBADgJxAWwJYQKYwLwwApJqYB0CGEIANgG4YAUA5NQIaUCuGDAlANwBQfVADMYdRCnQYuMAN58YMAPSKYAERAwKyDFAAWqMAHM%2BAXwGhIsFpRzxCkmAH4YARgBMAZhgAuGABYAVgA2fj4Ad31KLDF7TGk5BWU1DS0dfSNTPiA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oFtLlZkiACa1i0Dr0GoMAbXDYciaOOiQANPIVYF2SGQAWiMgGtkAYQ5MhlfJUsBDeNLD44iTdgC%2BHgLoaFeESkyBTUdAzMbJy0sExMRijI9tCEtGSW1raW6FBKKpDy3p5AA&tsconfig=&tokens=false) and I don't particularly want to give it more duplication.

But at the same time... The rule does have a entire option for this and it is highly specialised (just promises) so turning it on is less controversial than NUC.

**Comment by glentakahashi** [comment]
That other rule is very nice and catches the issue, however my problem with it is that it also requires setting `noUncheckedIndexAccess` = true for our code base too. It's very aggressive and has very little config options, so we can't say only enable it for `if` statements.

I suppose instead the better ticket to file would be to ask for more config options on the other rule and disable it for `?.`. Any thoughts?
